### ISSUE TITLE
import: URL clarification

### ISSUE THREAD

**Issue by guybedford** [WICG/import-maps#149]
The readme information on `import:` referrer handling is really excellent, but I just wanted to clarify one case for my own peace of mind which doesn't seem to be mentioned.

If I use an `import:` URL in a module specifier or dynamic import, will it still be resolved relative to the parent:

```js
import x from 'import:x';
import y from 'x';
const z = await import('import:x');

// Are x,y, z all equal?
```

In addition, does that mean a dynamic import in a script would be script-relative:

```html
<script src="/path/to/script.js"></script>
```

script.js
```js
import('./dep.js'); // loads /path/to/dep.js?
```

I would hope that the answer to the above would be yes for consistency, but it doesn't seem to be explicitly detailed from what I could see.

**Comment by guybedford** [comment]
Perhaps the above shouldn't be supported actually, since we then have the question of `import:x` maps themselves!?

Either way would be good to know!

**Comment by domenic** [comment]
Yeah, I guess we never did figure this out. The [initial import maps rewrite](https://github.com/WICG/import-maps/blob/fad2f0475b2d57a03d70461455e92a96fd9ea595/README.md#import-urls) had this as "TBD".

Making it fail certainly seems simplest. Maybe that would be a bit limiting, because you'll get a URL that sometimes you want to send to one place, and sometime send to `import()`?

I agree that if we were to allow it then we should basically just strip the leading `import:`.

**Comment by guybedford** [comment]
In theory,

<link rel=modulepreload href=import:import:x’>

Would attempt to resolve ‘import:x’ in the import map so that the default
behaviour (silent failure, and supporting mappings) may actually still be
desired!
